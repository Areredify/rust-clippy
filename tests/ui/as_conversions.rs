#![warn(clippy::as_conversions)]

macro_rules! mcr1 {
    (0_u32 as u64) => {
        ()
    };
}

macro_rules! mcr2 {
    () => {
        0_u32 as u64
    };
}

fn main() {
    let i = 0u32 as u64;

    let j = &i as *const u64 as *mut u64;

    let k = mcr1!(0_u32 as u64);

    let p = mcr2!();
}
