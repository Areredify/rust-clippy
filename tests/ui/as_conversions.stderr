error: using a potentially dangerous silent `as` conversion
  --> $DIR/as_conversions.rs:16:13
   |
LL |     let i = 0u32 as u64;
   |             ^^^^^^^^^^^
   |
   = note: `-D clippy::as-conversions` implied by `-D warnings`
   = help: consider using a safe wrapper for this conversion

error: using a potentially dangerous silent `as` conversion
  --> $DIR/as_conversions.rs:18:13
   |
LL |     let j = &i as *const u64 as *mut u64;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider using a safe wrapper for this conversion

error: using a potentially dangerous silent `as` conversion
  --> $DIR/as_conversions.rs:18:13
   |
LL |     let j = &i as *const u64 as *mut u64;
   |             ^^^^^^^^^^^^^^^^
   |
   = help: consider using a safe wrapper for this conversion

error: using a potentially dangerous silent `as` conversion
  --> $DIR/as_conversions.rs:11:9
   |
LL |         0_u32 as u64
   |         ^^^^^^^^^^^^
...
LL |     let p = mcr2!();
   |             ------- in this macro invocation
   |
   = help: consider using a safe wrapper for this conversion

error: aborting due to 4 previous errors

